netkan/netkan/metadata.py

```diff
@@ -121,3 +122,164 @@ def licenses(self):
             return lic
         else:
             return [lic]
+
+    class Version:
+
+        Pattern = re.compile("^(?:(?P<epoch>[0-9]+):)?(?P<version>.*)$")
+
+        def __init__(self, version_string):
+            self.string = version_string
+            match = self.Pattern.fullmatch(self.string)
+            if not match:
+                raise Exception('Invalid version format')
+            if match.group('epoch') and match.group('epoch').isnumeric():
+                self.epoch = int(match.group('epoch'))
+            else:
+                # https://github.com/KSP-CKAN/CKAN/blob/master/Spec.md#epoch
+                # 0 assumed if no epoch supplied
+                self.epoch = 0
+            self.bare_version = match.group('version')
+            if self.bare_version is None:
+                raise Exception
+
+        def __eq__(self, other):
+            return not self < other and not self > other
+
+        def __ne__(self, other):
+            return not self == other
+
+        def __lt__(self, other):
```

Comment:
The `@total_ordering` decoration from `functools` can apparently generate these stubs for us.

Reply:
It works fine for everything except `__eq__`, so I'll keep that one.

Reply:
Yup. :+1: 

> The class must define one of __lt__(), __le__(), __gt__(), or __ge__(). In addition, the class should supply an __eq__() method.

https://docs.python.org/2/library/functools.html